Add cameras and players to map by double clicking

hungnguyen94 opened the pull request:
This makes it possible to add cameras and players by double clicking on the map. It will popup a dialog, allowing you to select either a player or a camera. 
With this, we can omit the coordinates fields in the add player/camera dialog. 

I also moved the `uibModal` methods to a separate service.

ChrisBerg commented:
Nice work on the FabricMapDialogs!

ChrisBerg commented on src/main/webapp/app/map/map-table-view-template.html:
```diff
@@ -40,7 +40,7 @@
     </tr>
 
-    <tr class="btn btn-raised btn-xs" ng-click="vm.addCamera()">
+    <tr class="btn btn-raised btn-xs" ng-click="vm.addCamera(vm.project.map)">
```
The camera is now created in the corner where it almost falls of the map, could this be created somewhere else?

hungnguyen94 replied:
Are you adding cameras by clicking on the `add camera` button on the table? 
You can now add cameras by double clicking. The button on the table should probably be removed.

ChrisBerg commented on src/main/webapp/app/map/map-table-view-template.html:
```diff
@@ -83,7 +83,7 @@
     </tr>
 
-    <tr class="btn btn-raised btn-xs" ng-click="vm.addPlayer()">
+    <tr class="btn btn-raised btn-xs" ng-click="vm.addPlayer(vm.project.map)">
```
Same for map

ChrisBerg commented on src/main/webapp/app/map/fabric-map.directive.js:
```diff
@@ -77,7 +81,7 @@
+                var canvas = new fabric.CanvasEx(element[0]);
                 canvas.selection = false;
                 scope.canvas = canvas;
                 scope.canvas.add(label);
```
Could this label be removed on mouse click? It now stays at the old position when dragging a camera or a player, which seems kind of odd.

hungnguyen94 replied:
Ideally, the label should move with the camera or player. But for now, I'll see if I can remove it.

ChrisBerg commented:
Works really well, nice work! One small remark: could you set the focus of the add camera dialog and add player dialog form to the "Name" field?

This could be done for example at the `CameraDialogController` by changing `angular.element('.form-group:eq(1)>input').focus();` to `angular.element('.form-group:eq(0)>input').focus();` (line 16 in /app/map/camera-dialog.controller.js)

trombomart commented:
Nothing to comment besides Chris' comments. Nice job!

hungnguyen94 commented:
`index.html` is actually needed here, because the `fabric.canvasex.js` file is not injected by `gulp`, but has to be manually loaded. I don't commit unnecessary files :wink:

cadw commented:
Looks great to me, it is mergeable in my eyes.

hungnguyen94 commented:
Label moves with the cameras and players now. I changed the focus of the camera dialog to the `name` field. I also removed the `Add player` and `Add camera` button from the table view.

hungnguyen94 commented:
![image](https://cloud.githubusercontent.com/assets/9165947/16147093/9acd0284-3481-11e6-8969-52ea95199735.png)
Highlighting is made a bit more clear.

ChrisBerg commented:
@hungnguyen94 nice, i am getting real thirsty right now...

hungnguyen94 commented:
So many merge commits :cry:

ChrisBerg commented:
Nice, i think it is mergeable after all checks pass.
